src/components/icon/icon.styles.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css, keyframes } from '@emotion/react';
+import { logicalCSS, euiCanAnimate } from '../../global_styling';
+import { UseEuiTheme } from '../../services';
+import { EuiIconProps } from './icon';
+
+const _iconSize = (size: string) => {
+  return `
+    ${logicalCSS('width', size)};
+    ${logicalCSS('height', size)};
+  `;
+};
+
+const _iconLoadingOpacity = 0.05;
+
+const _iconLoading = keyframes`
+  0% {
+    opacity: ${_iconLoadingOpacity};
+  }
+
+  100% {
+    opacity: 1;
+  }
+`;
+
+export const euiIconStyles = (
+  { euiTheme }: UseEuiTheme,
+  color: EuiIconProps['color']
+) => ({
+  // Base
+  euiIcon: css`
+    flex-shrink: 0; // Ensures it never scales down below its intended size
+    display: inline-block;
+    vertical-align: middle;
+    fill: currentColor;
+    transform: translate(0, 0); // Hack to fix Firefox "softness"
+
+    &:focus {
+      opacity: 1; // We often hide icons on hover. Make sure they appear on focus.
+    }
+
+    // Use the negative fill on SVG elements that need to flip portions
+    // of the icon to light and dark compatible modes. You can see this
+    // in use on the Elastic logo icons.
+    .euiIcon__fillNegative {
```

Comment:
I was originally going to ask that we flatten this CSS but it looks like we can't because the class names are hard-coded into our SVGs. I think it's worth noting that in a comment
```suggestion
    // Note: This className is hard-coded into our SVGs and cannot be converted to Emotion
    .euiIcon__fillNegative {
      // Use the negative fill on SVG elements that need to flip portions
      // of the icon to light and dark compatible modes. You can see this
      // in use on the Elastic logo icons.
```